x-pack/platform/plugins/shared/onechat/server/utils/kibana_mcp_http_transport.ts

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Transport } from '@modelcontextprotocol/sdk/shared/transport.js';
+import type { JSONRPCMessage, RequestId } from '@modelcontextprotocol/sdk/types.js';
+import type { KibanaResponseFactory, KibanaRequest } from '@kbn/core-http-server';
+import {
+  isJSONRPCRequest,
+  isJSONRPCResponse,
+  isJSONRPCError,
+  JSONRPCMessageSchema,
+  isInitializeRequest,
+} from '@modelcontextprotocol/sdk/types.js';
+import { Logger } from '@kbn/logging';
+import { AuthInfo } from '@modelcontextprotocol/sdk/server/auth/types';
+import { randomUUID } from 'node:crypto';
+import type { IKibanaResponse } from '@kbn/core/server';
+
+// SDK error codes
+export const CONNECTION_CLOSED = -32000;
+
+// Standard JSON-RPC error codes
+export const PARSE_ERROR = -32700;
+export const INVALID_REQUEST = -32600;
+export const METHOD_NOT_FOUND = -32601;
+export const INVALID_PARAMS = -32602;
+export const INTERNAL_ERROR = -32603;
+
+/**
+ * Server transport for Streamable HTTP: this implements the MCP Streamable HTTP transport specification.
+ * It supports direct HTTP responses. It doesn't support SSE streaming.
+ *
+ * It is compatible with Kibana's HTTP request/response abstractions. Implementation is adapted from:
+ * https://github.com/modelcontextprotocol/typescript-sdk/blob/main/src/server/streamableHttp.ts
+ *
+ * In stateful mode:
+ * - Session ID is generated and included in response headers
+ * - Session ID is always included in initialization responses
+ * - Requests with invalid session IDs are rejected with 404 Not Found
+ * - Non-initialization requests without a session ID are rejected with 400 Bad Request
+ * - State is maintained in-memory (connections, message history)
+ *
+ * In stateless mode:
+ * - No Session ID is included in any responses
+ * - No session validation is performed
```

Comment:
I assume this is coming from the transport implementation you forked? From my understanding we only use stateless mode, right? Any upside keeping both atm?